lib/identity_cache.rb

```diff
@@ -94,8 +94,9 @@ def should_fill_cache? # :nodoc:
     end
 
     def should_use_cache? # :nodoc:
-      pool = ActiveRecord::Base.connection_pool
-      !pool.active_connection? || pool.connection.open_transactions == 0
+      ActiveRecord::Base.connection_handler.connection_pool_list.none? do |pool|
+        pool.active_connection? && pool.connection.current_transaction.joinable?
```

Comment:
It seems like this would normally work, because non-joinable transactions is an obscure feature that internally is only used for the console sandbox and transactional fixtures.  However, that same obscurity of the feature makes this code hard to understand, along with relying on current_transaction to return NullTransaction.

Could you comment the code so that it can be read without also reading the active record internals?